### ISSUE TITLE
error message when trying to delete specific player

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#1289]
on the QMU instance, i have successfully deleted a lot of players, but there are two accounts, both belonging to teddy, which i can't delete. this is the error message i get:

(psycopg2.errors.ForeignKeyViolation) update or delete on table "upstage_user" violates foreign key constraint "scene_owner_id_fkey" on table "scene" DETAIL: Key (id)=(123) is still referenced from table "scene". [SQL: DELETE FROM upstage_user WHERE upstage_user.id = %(id_1)s] [parameters: {'id_1': '123'}] (Background on this error at: https://sqlalche.me/e/14/gkpj)

note: this issue applies to all instances, not just QMU

**Comment by gloriajw** [comment]
This means a scene still belongs to him, and you need to reassign it or delete it first.

**Comment by helenvarley** [comment]
ah! thank you - it worked now. is it possible to have a meaningful message to that effect? i didn't understand that from the error message it gave.

**Comment by vickis** [comment]
This is the same as the one I reported earlier - where I found and deleted an old Stage which contained a scene - in that case the user was James :) 

13 Jun 2024 at 01:34 by ***@***.***:

**Comment by helenvarley** [comment]
yes, but we're keeping this issue open as it needs a meaningful error message so that anyone encountering it will get the info they need to sort it out :)

**Comment by helenvarley** [comment]
the error message should say:

"The player you are trying to delete still owns a stage or media items on this UpStage instance. Please use the Studio to find these and either delete or assign to other players; then you will be able to delete this player."

(i don't think "scenes" are able to belong to a specific player, are they? there is no information anywhere about scenes belonging to anyone. i can see that it does say "scene" in the error message, but does this actually mean "scenes" - as in the scenes we can save on a stage - or "stages" - which are owned by a specific player?)

**Comment by webdev604** [comment]
hi @gloriajw can you send the error messages on backend?

**Comment by vickis** [comment]
The scenes (like the stage would have to 'belong' to someone so I guess it would be by default whoever created them
perhaps the assignee for scenes should be the stage (if that is possible)

**Comment by helenvarley** [comment]
this is still not resolved. when i try to delete a player, i first get the popup message: "Deleting this user will also delete all of this user's stages. All of this user's media will belong to you. This cannot be undone! Are you sure you want to continue?" and when i click "yes", i get the error message above.

since the first message tells me that the player's stages will be deleted and any media will be transferred to me, it's not consistent to then get a message telling me that i have to go and manually locate and delete or reassign any stages or media. 

furthermore, there is no way to change ownership of either stages or media to another player. so the instruction is impossible ...

**Comment by helenvarley** [comment]
testing this by creating a new player, zecica. i created the player and then deleted and it worked.

then i created another new player, zecica2, logged in, created a stage and uploaded some media. i assigned it to my stage, and saved a screne on the stage. 

then (as helen, admin account) i tried to delete zecica2. i got this error:

(psycopg2.errors.ForeignKeyViolation) update or delete on table "stage" violates foreign key constraint "scene_stage_id_fkey" on table "scene" DETAIL: Key (id)=(34) is still referenced from table "scene". [SQL: DELETE FROM stage WHERE stage.owner_id = %(owner_id_1)s] [parameters: {'owner_id_1': '47'}] (Background on this error at: https://sqlalche.me/e/20/gkpj)

**Comment by nnguyenquangg** [comment]
it's fixed @helenvarley

**Comment by helenvarley** [comment]
i'm still getting this error when i try to delete the account zecica2:

(psycopg2.errors.ForeignKeyViolation) update or delete on table "stage" violates foreign key constraint "scene_stage_id_fkey" on table "scene" DETAIL: Key (id)=(34) is still referenced from table "scene". [SQL: DELETE FROM stage WHERE stage.owner_id = %(owner_id_1)s] [parameters: {'owner_id_1': '47'}] (Background on this error at: https://sqlalche.me/e/20/gkpj)

![Image](https://github.com/user-attachments/assets/34529526-8601-4b6d-9cc6-8e3e4f1a5bea)

**Comment by helenvarley** [comment]
i created a new zecica1 player, and was able to delete this - but it did not have any media or stages associated with it, i just created it and then immediately deleted it. this worked.

**Comment by helenvarley** [comment]
i was also able to delete the player vicki, who i created 11 days ago, and who also didn't own any media or stages. so it seems like the problem may be specific to deleting a player who has media and/or stages.

**Comment by nnguyenquangg** [comment]
The error happens when deleting users, which has scenes
Pls check again @helenvarley

**Comment by nnguyenquangg** [comment]
It's possible @helenvarley

**Comment by nnguyenquangg** [comment]
it's fixed @helenvarley

**Comment by helenvarley** [comment]
great! i tested this by creating a player zecica, and as zecica creating a stage and uploading some media. i also gave the media items different permission statuses to see if those would be carried over.

then as helen (Admin), i deleted the player zecica. i got the warning, that zecica's stages would be deleted, & that zecica's media would belong to me. i clicked ok. the player zecica disappeared from the player list, zecica's stage disappeared from the stages list, and the media is in the media list showing as belonging to helen.

so this is all working as i expect. thank you! 🙏
